File: download/src/main/java/com/huison/download/utils/HandlerUtil.java
package com.huison.download.utils;

import android.os.Handler;
import android.os.Looper;

/**
 * Created by huisonma on 2019/5/15.
 */
public class HandlerUtil {

    private static final Handler sUiHandler = new Handler(Looper.getMainLooper());

    public static void postOnUiThread(Runnable runnable) {
        sUiHandler.post(runnable);
    }

    public static void postOnUiThreadDelayed(Runnable runnable, long delayMillis) {
        sUiHandler.postDelayed(runnable, delayMillis);
    }

    public static void remove(Runnable runnable) {
        sUiHandler.removeCallbacks(runnable);
    }
}

File: ktdownload/src/main/java/com/huison/kotlin/download/utils/HandlerUtil.kt
package com.huison.kotlin.download.utils

import android.os.Handler
import android.os.Looper

/**
 * Created by huisonma 2019/9/3.
 */
class HandlerUtil {

    companion object {

        private val sUiHandler: Handler = Handler(Looper.getMainLooper())

        @JvmStatic
        fun postOnUiThread(runnable: Runnable) {
            sUiHandler.post(runnable)
        }

        @JvmStatic
        fun postOnUiThreadDelay(runnable: Runnable, delayMillis: Long) {
            sUiHandler.postDelayed(runnable, delayMillis)
        }

        @JvmStatic
        fun remove(runnable: Runnable) {
            sUiHandler.removeCallbacks(runnable)
        }
    }
}